### ISSUE TITLE
Can't turn off one of the Govee H7020 outdoor lights

### ISSUE THREAD

**Issue by nawwman** [wez/govee-lan-hass#49]
I have 2 outdoor lights of the H7020 model. Unfortunately pure LAN control is not possible for them, so I had to go the WiFi route.

For whatever reason, one of the outdoor lights (called "Loorbeer links") stopped responding to any commands, so I can't turn it on or off or change colors. The other light responds without a problem. I've turned on debug mode in the intergration, and when I send the turn off command to that light here's what I can see in the log viewer:

> 2023-12-27 10:42:46.031 DEBUG (MainThread) [custom_components.govee_lan.light] turn OFF 52:81:D2:37:34:32:28:8F light.lichter_loorbeer_links with {}
> 
> 2023-12-27 10:42:46.777 DEBUG (MainThread) [govee_led_wez.http] http control request {'device': '52:81:D2:37:34:32:28:8F', 'model': 'H7020', 'cmd': {'name': 'turn', 'value': 'off'}} -> <ClientResponse(https://developer-api.govee.com/v1/devices/control) [200 OK]>
> 
> 2023-12-27 10:42:46.783 DEBUG (MainThread) [custom_components.govee_lan.light] device state updated: 52:81:D2:37:34:32:28:8F entity_id=light.lichter_loorbeer_links --> {'turned_on': False, 'brightness_pct': 100, 'color': GoveeColor(red=255, green=35, blue=211), 'color_temperature': None} {'device_id': '52:81:D2:37:34:32:28:8F', 'model': 'H7020', 'http_definition': GoveeHttpDeviceDefinition(device_id='52:81:D2:37:34:32:28:8F', model='H7020', device_name='Lichter Loorbeer links', controllable=True, retrievable=True, supported_commands=['turn', 'brightness', 'color', 'colorTem'], properties={'colorTem': {'range': {'min': 2000, 'max': 9000}}}), 'state': {'turned_on': False, 'brightness_pct': 100, 'color': GoveeColor(red=255, green=35, blue=211), 'color_temperature': None}}
> 
> 2023-12-27 10:42:46.803 DEBUG (MainThread) [custom_components.govee_lan.light] skip async_update for {'device_id': '52:81:D2:37:34:32:28:8F', 'model': 'H7020', 'http_definition': GoveeHttpDeviceDefinition(device_id='52:81:D2:37:34:32:28:8F', model='H7020', device_name='Lichter Loorbeer links', controllable=True, retrievable=True, supported_commands=['turn', 'brightness', 'color', 'colorTem'], properties={'colorTem': {'range': {'min': 2000, 'max': 9000}}}), 'state': {'turned_on': False, 'brightness_pct': 100, 'color': GoveeColor(red=255, green=35, blue=211), 'color_temperature': None}} light.lichter_loorbeer_links as elapsed 1 < 600
> 
> 2023-12-27 10:43:00.432 DEBUG (MainThread) [custom_components.govee_lan.light] skip async_update for {'device_id': '52:81:D2:37:34:32:28:8F', 'model': 'H7020', 'http_definition': GoveeHttpDeviceDefinition(device_id='52:81:D2:37:34:32:28:8F', model='H7020', device_name='Lichter Loorbeer links', controllable=True, retrievable=True, supported_commands=['turn', 'brightness', 'color', 'colorTem'], properties={'colorTem': {'range': {'min': 2000, 'max': 9000}}}), 'state': {'turned_on': False, 'brightness_pct': 100, 'color': GoveeColor(red=255, green=35, blue=211), 'color_temperature': None}} light.lichter_loorbeer_links as elapsed 14 < 600
> 
> 2023-12-27 10:43:00.434 DEBUG (MainThread) [custom_components.govee_lan.light] skip async_update for {'device_id': '9A:A4:D0:30:39:39:19:4D', 'model': 'H7020', 'http_definition': GoveeHttpDeviceDefinition(device_id='9A:A4:D0:30:39:39:19:4D', model='H7020', device_name='Lichter Feige diagonal', controllable=True, retrievable=True, supported_commands=['turn', 'brightness', 'color', 'colorTem'], properties={'colorTem': {'range': {'min': 2000, 'max': 9000}}}), 'state': {'turned_on': True, 'brightness_pct': 100, 'color': GoveeColor(red=28, green=12, blue=255), 'color_temperature': None}} light.lichter_feige_diagonal as elapsed 109 < 600
> 
> 2023-12-27 10:43:30.438 DEBUG (MainThread) [custom_components.govee_lan.light] skip async_update for {'device_id': '52:81:D2:37:34:32:28:8F', 'model': 'H7020', 'http_definition': GoveeHttpDeviceDefinition(device_id='52:81:D2:37:34:32:28:8F', model='H7020', device_name='Lichter Loorbeer links', controllable=True, retrievable=True, supported_commands=['turn', 'brightness', 'color', 'colorTem'], properties={'colorTem': {'range': {'min': 2000, 'max': 9000}}}), 'state': {'turned_on': False, 'brightness_pct': 100, 'color': GoveeColor(red=255, green=35, blue=211), 'color_temperature': None}} light.lichter_loorbeer_links as elapsed 44 < 600
> 
> 2023-12-27 10:43:30.441 DEBUG (MainThread) [custom_components.govee_lan.light] skip async_update for {'device_id': '9A:A4:D0:30:39:39:19:4D', 'model': 'H7020', 'http_definition': GoveeHttpDeviceDefinition(device_id='9A:A4:D0:30:39:39:19:4D', model='H7020', device_name='Lichter Feige diagonal', controllable=True, retrievable=True, supported_commands=['turn', 'brightness', 'color', 'colorTem'], properties={'colorTem': {'range': {'min': 2000, 'max': 9000}}}), 'state': {'turned_on': True, 'brightness_pct': 100, 'color': GoveeColor(red=28, green=12, blue=255), 'color_temperature': None}} light.lichter_feige_diagonal as elapsed 139 < 600

It's like it's not receiving any commands, but also not throwing any kind of error. Also what's up with the ongoing debug commands about skipping updates?

and the attributes of the misbehaving outdoor lights:

> min_color_temp_kelvin: 2000 
> max_color_temp_kelvin: 9000 
> min_mireds: 111 
> max_mireds: 500 
> supported_color_modes: brightness, color_temp, rgb 
> color_mode: null 
> brightness: null 
> color_temp_kelvin: null 
> color_temp: null 
> hs_color: null 
> rgb_color: null 
> xy_color: null 
> http_enabled: true 
> ble_enabled: false 
> lan_enabled: false 
> update_status: ok at Wed Dec 27 10:33:00 2023 
> timeout_count: 0 
> friendly_name: Lichter Loorbeer links s
> upported_features: 0

so it does seem like it's getting data updates provided to it, but not the actual numbers and attributes

**Comment by Tom-Sumner** [comment]
in the attributes it says lan_enabled: false. have you tried turning that on in the govee app?
and are you lights up to date with most recent firmware?

**Comment by wez** [comment]
You need to keep in mind that Govee devices aren't super great from a quality perspective; I would suggest physically power cycling the device that isn't responding.

When using the HTTP API, you are sending requests to a server in the cloud that will then use (probably) MQTT to publish a message to the device via a broker. There is no direct connection from HASS to the device in that situation, neither is there a direct connection from govee to the device.  The device itself will asynchronously push a notification back over mqtt with its state.

With the LAN API the HASS integration can directly send a request to the device, but using UDP, an unreliable protocol. There is no direct confirmation about the success of any given send. The HASS integration also asks it to report its status, again, using the same unreliable protocol. The device may then choose to send that status.

What shows up in the hass logs as a result of this is not necessarily very useful beyond capturing what the integration thinks about the state of the device.

Regarding skipping updates in the logs: Govee has fairly restrictive API limits for its cloud API so the integration tries to avoid polling it too frequently, so that you have quota left to actually control the device.

The fact that you see a last update time show for the device just means that the integration was able to talk to Govee's cloud service and get a non-error response.  Because there isn't a direct link between the request for status and actually checking the device, and because the cloud service itself can report cached status, it doesn't necessarily mean anything useful.

**Comment by nawwman** [comment]
> in the attributes it says lan_enabled: false. have you tried turning that on in the govee app? and are you lights up to date with most recent firmware?

Oh right you are. I've logged in to the app itself and for whatever reason, one of the lights didn't have any WiFi connection (didn't even have log in data). Once I've logged it again I could trigger it via HA

@wez yeah this is kind of hindsight, I'm just dealing with the bulbs that I've already bought. If I could do it again, I'd buy some outdoor bulbs which HA could talk to directly without having to go through an internet cloud. Hell some of Govee's own devices seem to have a pure LAN mode, just no these particular light bulbs unfortunately
